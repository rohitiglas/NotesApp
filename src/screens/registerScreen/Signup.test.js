import 'react-native';
import React from 'react';
import {Alert} from 'react-native';
import SignUp from './Signup';
import {
  render,
  cleanup,
  fireEvent,
  flushMicrotasksQueue,
} from '@testing-library/react-native';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
let mock = new MockAdapter(axios);
jest.mock('react-native/Libraries/Alert/Alert', () => ({
  alert: jest.fn(),
}));
describe('Signup Form', () => {
  const fakeNavigation = {
    navigate: jest.fn(),
  };
  let wrapper;
  beforeEach(() => {
    wrapper = render(<SignUp navigation={fakeNavigation} />);
  });
  afterEach(() => {
    cleanup();
    wrapper = null;
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show Form Empty error on button click ', () => {
    const {getByTestId} = wrapper;
    getByTestId('input-email');
    getByTestId('input-username');
    getByTestId('input-password');
    getByTestId('input-c-password');
    const button = getByTestId('submit-button');
    fireEvent.press(button);
    const error = getByTestId('text-error');
    expect(error.props.children).toBe('Form fields cannot be blank');
  });
  it('should show error when email is empty', () => {
    const {getByTestId} = wrapper;
    const email = getByTestId('input-email');
    const password = getByTestId('input-password');
    const button = getByTestId('submit-button');
    fireEvent.changeText(password, '12345678');
    fireEvent.press(button);
    expect(email.props.error).toBe('Email cannot be blank');
  });
  it('should show error when UserName is empty', () => {
    const {getByTestId} = wrapper;
    const email = getByTestId('input-email');
    const userName = getByTestId('input-username');
    const password = getByTestId('input-password');
    const button = getByTestId('submit-button');
    fireEvent.changeText(email, 'abc@gmail.com');
    fireEvent.changeText(password, '12345678');
    fireEvent.press(button);
    expect(userName.props.error).toBe('Username cannot be blank');
  });
  it('should show error when Password is empty', () => {
    const {getByTestId} = wrapper;
    const email = getByTestId('input-email');
    const userName = getByTestId('input-username');
    const password = getByTestId('input-password');
    const button = getByTestId('submit-button');
    fireEvent.changeText(email, 'abc@gmail.com');
    fireEvent.changeText(userName, 'Rohit');
    fireEvent.press(button);
    expect(password.props.error).toBe('Password cannot be blank');
  });
  it('should show error when confirm password empty', () => {
    const {getByTestId} = wrapper;
    const email = getByTestId('input-email');
    const userName = getByTestId('input-username');
    const password = getByTestId('input-password');
    const cPassword = getByTestId('input-c-password');
    const button = getByTestId('submit-button');
    fireEvent.changeText(email, 'abc@gmail.com');
    fireEvent.changeText(userName, 'Rohit');
    fireEvent.changeText(password, '123456');
    fireEvent.changeText(cPassword, '');
    fireEvent.press(button);
    expect(cPassword.props.error).toBe('Password cannot be blank');
  });

  it('should open Login Screen on login button click', () => {
    const {getByTestId} = wrapper;
    const button = getByTestId('login-button');
    fireEvent.press(button);
    expect(fakeNavigation.navigate).toBeCalledWith('Login');
  });
  it('should login api called on submit button', async () => {
    mock.onPost('/api/registerUser').reply(200, {});
    const {getByTestId, queryByTestId} = wrapper;
    const email = getByTestId('input-email');
    const userName = getByTestId('input-username');
    const password = getByTestId('input-password');
    const cPassword = getByTestId('input-c-password');
    const button = getByTestId('submit-button');
    fireEvent.changeText(email, 'abc@gmail.com');
    fireEvent.changeText(userName, 'Rohit');
    fireEvent.changeText(password, '123456');
    fireEvent.changeText(cPassword, '123456');
    fireEvent.press(button);
    await fireEvent.press(button);
    await flushMicrotasksQueue();
    expect(mock.history.post.length).toBe(2);
    expect(mock.history.post[0].data).toBe(
      JSON.stringify({
        email: 'abc@gmail.com',
        userName: 'Rohit',
        password: '123456',
      }),
    );
    const error = queryByTestId('text-error');
    expect(error).toBe(null);
    const alertSpy = jest.spyOn(Alert, 'alert');
    expect(alertSpy).toHaveBeenCalled();
  });
  it('should show error when login api failed on submit button', async () => {
    mock.onPost('/api/registerUser').reply(500, {});
    const {getByTestId, queryByTestId} = wrapper;
    const email = getByTestId('input-email');
    const userName = getByTestId('input-username');
    const password = getByTestId('input-password');
    const cPassword = getByTestId('input-c-password');
    const button = getByTestId('submit-button');
    fireEvent.changeText(email, 'abc@gmail.com');
    fireEvent.changeText(userName, 'Rohit');
    fireEvent.changeText(password, '123456');
    fireEvent.changeText(cPassword, '123456');
    fireEvent.press(button);
    await fireEvent.press(button);
    await flushMicrotasksQueue();
    expect(mock.history.post.length).toBe(4);
    expect(mock.history.post[0].data).toBe(
      JSON.stringify({
        email: 'abc@gmail.com',
        userName: 'Rohit',
        password: '123456',
      }),
    );
    const error = queryByTestId('text-error');
    expect(error).toBe(null);
    const alertSpy = jest.spyOn(Alert, 'alert');
    expect(alertSpy).toHaveBeenCalled();
  });
});
