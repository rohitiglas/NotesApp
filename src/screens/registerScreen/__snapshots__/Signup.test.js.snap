// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Form renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID="loader"
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </Modal>
  <View
    style={
      Object {
        "marginLeft": 25,
        "marginRight": 25,
        "marginTop": 80,
      }
    }
  >
    <View>
      <TextInput
        allowFontScaling={true}
        error=""
        inputContainerStyle={
          Object {
            "borderBottomWidth": 0,
          }
        }
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#808080"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#1e1e1d",
            "borderColor": "#FFF",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#FFF",
            "fontSize": 18,
            "fontWeight": "500",
            "padding": 20,
          }
        }
        testID="input-email"
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "color": "#ff2251",
            "fontSize": 14,
            "fontWeight": "900",
            "margin": 10,
          }
        }
      >
         
        
         
      </Text>
    </View>
    <View>
      <TextInput
        allowFontScaling={true}
        error=""
        inputContainerStyle={
          Object {
            "borderBottomWidth": 0,
          }
        }
        onChangeText={[Function]}
        placeholder="UserName"
        placeholderTextColor="#808080"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#1e1e1d",
            "borderColor": "#FFF",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#FFF",
            "fontSize": 18,
            "fontWeight": "500",
            "padding": 20,
          }
        }
        testID="input-username"
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "color": "#ff2251",
            "fontSize": 14,
            "fontWeight": "900",
            "margin": 10,
          }
        }
      >
         
        
         
      </Text>
    </View>
    <View>
      <TextInput
        allowFontScaling={true}
        error=""
        inputContainerStyle={
          Object {
            "borderBottomWidth": 0,
          }
        }
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#808080"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#1e1e1d",
            "borderColor": "#FFF",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#FFF",
            "fontSize": 18,
            "fontWeight": "500",
            "padding": 20,
          }
        }
        testID="input-password"
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "color": "#ff2251",
            "fontSize": 14,
            "fontWeight": "900",
            "margin": 10,
          }
        }
      >
         
        
         
      </Text>
    </View>
    <View>
      <TextInput
        allowFontScaling={true}
        error=""
        inputContainerStyle={
          Object {
            "borderBottomWidth": 0,
          }
        }
        onChangeText={[Function]}
        placeholder="Confirm Password"
        placeholderTextColor="#808080"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#1e1e1d",
            "borderColor": "#FFF",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#FFF",
            "fontSize": 18,
            "fontWeight": "500",
            "padding": 20,
          }
        }
        testID="input-c-password"
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "color": "#ff2251",
            "fontSize": 14,
            "fontWeight": "900",
            "margin": 10,
          }
        }
      >
         
        
         
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ff2251",
          "borderRadius": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 20,
          "opacity": 1,
        }
      }
      testID="submit-button"
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 22,
            "fontWeight": "700",
            "paddingBottom": 10,
            "paddingLeft": 30,
            "paddingRight": 30,
            "paddingTop": 10,
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "marginLeft": 10,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Already a member of Notes?
      </Text>
      <Text
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "color": "#ff2251",
            "fontSize": 16,
            "fontWeight": "900",
            "marginLeft": 8,
            "marginRight": 10,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
        testID="login-button"
      >
        Login
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
